### ISSUE TITLE
Add `aiter` and `anext` builtin function

### ISSUE THREAD

**Issue by youknowone** [RustPython/RustPython#3609]
## Feature

- [ ] `builtins.aiter`: https://docs.python.org/3/library/functions.html#aiter
- [ ] `builtins.anext`: https://docs.python.org/3/library/functions.html#anext

**Comment by Snowapril** [comment]
`builtins.aiter` can be implemented quite easy with just a few codes.
but for `builtins.anext`, it seems `anext_awaitable` must be implemented first.
```c
static PyObject *
builtin_anext_impl(PyObject *module, PyObject *aiterator,
                   PyObject *default_value)
{
    // ...skip
    awaitable = (*t->tp_as_async->am_anext)(aiterator);
    if (default_value == NULL) {
        return awaitable;
    }

    PyObject* new_awaitable = PyAnextAwaitable_New(
            awaitable, default_value);
    Py_DECREF(awaitable);
    return new_awaitable;
}
```

**Comment by sum12** [comment]
Thanks for the "good for issue". I am probably misunderstanding things. 
the slots for `tp_as_async->am_aiter` is missing. According to cpython that would be the correct way to check if the object has `__aiter__` implementaed, Is this the correct way to proceed, probably an outline of how this could be done would be great

**Comment by youknowone** [comment]
@sum12 you are right. that's the right way. But instead of the right way, you can implement it by calling `__aiter__` directly.

If you are interested in the right way, You can do it by looking at other slots, like `IterNext` or `Callable`. The right way would not be the 'good first issue'.

**Comment by youknowone** [comment]
Thanks @sum12 and @YashSuthar983 ! this is done!

Thanks @Snowapril for advices
